ui/src/app/utils/runtime-hooks.tsx

```diff
@@ -440,3 +442,31 @@ export const withRuntimeStore = () => (WrappedComponent) => {
     return <WrappedComponent {...props} runtimeStore={value} />;
   };
 };
+
+/**
+ * Hook that initializes the runtime and disk stores and returns their loading states.
+ * Use this to show loading indicators while runtime and disk data are being fetched.
+ *
+ * @param workspaceNamespace - The namespace of the workspace
+ * @returns Object containing loading states for runtime and disk
+ */
+export const useRuntimeAndDiskStores = (workspaceNamespace: string) => {
+  // Initialize runtime and disk stores
+  useRuntime(workspaceNamespace);
```

Comment:
I suspect that the answer is "it's too complicated to get into today" but in your recent research, have you found a good way to distinguish what "useRuntime" does in contrast to "useStore(runtimeStore")?  Some renaming or additional comments could be very helpful here.

Reply:
same Q for disks